.github/workflows/deploy.yml

```diff
@@ -0,0 +1,163 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - master
+
+  # Run actions on PRs, but only deploy on master
+  pull_request:
+
+jobs:
+  cargo_docs:
+    name: Cargo Docs
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          toolchain: stable
+          profile: minimal
+          components: rust-docs
+          override: true
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: doc-${{ runner.os }}-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
+
+      - run: make doc
+
+      - name: Deploy
+        if: github.ref == 'refs/heads/master'
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: target/doc
+
+  event_schema:
+    name: Event Schema
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: schema-${{ runner.os }}-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Generate Schema
+        uses: actions-rs/cargo@v1
+        with:
+          command: run
+          args: -p generate-schema -- -o event.schema.json
+
+      - name: Deploy
+        if: github.ref == 'refs/heads/master'
+        env:
+          GITHUB_TOKEN: ${{ secrets.DATA_SCHEMAS_GITHUB_TOKEN }}
+        run: |
+          git config --global user.name "$(git log -1 --pretty=format:%an $GITHUB_SHA)"
+          git config --global user.email "$(git log -1 --pretty=format:%ae $GITHUB_SHA)"
+          git clone https://getsentry-bot:$GITHUB_TOKEN@github.com/getsentry/sentry-data-schemas
```

Comment:
I think this part can/should be replaced with `actions/checkout`

Reply:
Sounds good, I'll consider that in a follow-up PR. These jobs were just moved, so I would like to not touch them. 

My plan is to make this a reusable action that we can use for all sorts of cross-committing activities we do in the organization.